lib/ukfallocbuddy/tests/test_ukfallocbuddy.c

```diff
@@ -0,0 +1,179 @@
+#include <uk/test.h>
+#include <uk/fallocbuddy.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+
+struct uk_falloc *generate(void *fa_meta, void *start_zone, unsigned long len, __paddr_t *new_start);
+
+
+UK_TESTCASE(ukfallocbuddy, test_init)
+{
+	// test if the init function works
+	struct uk_falloc *fa;
+
+	fa = malloc(100000);
+	UK_TEST_EXPECT_SNUM_EQ(uk_fallocbuddy_init(fa), 0);
+	free(fa);
+}
+
+UK_TESTCASE(ukfallocbuddy, test_size)
+{
+	UK_TEST_EXPECT_SNUM_EQ(uk_fallocbuddy_size(), 368);
```

Comment:
Is this always going to be 368?

Reply:
It's true. The result is not constant. I will remove it.